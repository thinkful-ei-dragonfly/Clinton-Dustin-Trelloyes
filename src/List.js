import React from 'react';
import Card from './Card';


function List(props){
  
  //.map over an array to create JSX content
  return (
    <section className="List">
      <header className="List-header">
        <h2>{props.header}</h2>
      </header>
      <div className ="List-cards">
        {props.cards.map((card) => 
          <Card 
            title={card.title}
            content={card.content}
            />      
          )}
      </div>
    </section>
  );
}