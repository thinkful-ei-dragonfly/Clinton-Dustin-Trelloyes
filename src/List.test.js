
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';


describe('List Component', () => {

  it('Renders without Crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<List cards={[{ title: 'Twelth card', content: 'lorem ipsum' },
    {title: 'Thirteenth card', content: 'lorem ipsum' }]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders UI as expected', () =>{
    const tree = renderer
      .create(<List 
        key = {0}
        header={'header'}
        cards={[{ title: 'Twelth card', content: 'lorem ipsum' },
              {title: 'Thirteenth card', content: 'lorem ipsum' }]}
            />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  });

});
