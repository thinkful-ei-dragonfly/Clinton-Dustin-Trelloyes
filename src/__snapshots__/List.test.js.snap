// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      header
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        Twelth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Thirteenth card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
